ENH: Add GitHub Actions configuration

thewtex opened the pull request.

thewtex committed:
ENH: Add GitHub Actions configuration

thewtex committed:
COMP: Update GTest TYPED_TEST_CASE to TYPED_TEST_SUITE

From the warning:

  /home/matt/src/ITKBoneEnhancement/test/itkDescoteauxEigenToMeasureParameterEstimationFilterUnitTest.cxx:149:1:
  warning: 'TypedTestCaseIsDeprecated' is deprecated: TYPED_TEST_CASE is
  deprecated, please use TYPED_TEST_SUITE [-Wdeprecated-declarations]
  TYPED_TEST_CASE(itkDescoteauxEigenToMeasureParameterEstimationFilterUnitTest,
  TestingLabelTypes);

dzenanz reviewed (approved):
What is the difference between them?

thewtex committed:
DOC: Add missing Doxygen \ingroup BoneEnhancement

thewtex commented:
> What is the difference between them?

The old one is deprecated :-P. I assume SUITE is a more appropriate name.

thewtex committed:
BUG: Remove ald Azure Pipelines configuration

thewtex committed:
BUG: Double parallelism in EigenToMeasureImageFilter

DynamicThreadedGenerateData calls ParallelizeImageRegion, but we were
again calling ParallelizeImageRegion. This was resulting in a deadlock
of itkDescoteauxEigenToMeasureImageFilterUnitTest.

thewtex committed:
COMP: Update EigenValueOrder enum using declaration

To address::

  include/itkMultiScaleHessianEnhancementImageFilter.h:129:53: warning: declaration 'using EigenValueOrderEnum = enum class itk::SymmetricEigenAnalysisEnums::EigenValueOrder' does not declare anything

thewtex committed:
BUG: Retain previous EigenToMeasureImageFilter

thewtex committed:
DOC: Add Python package installation instructions

thewtex committed:
DOC: Add Python package description
